File: app/src/main/java/com/nickand/pokemondex/MainActivity.java
package com.nickand.pokemondex;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class MainActivity extends AppCompatActivity {
    private PokeApiService pokeApiService;
    private RecyclerView recyclerView;
    private PokemonListAdapter adapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        recyclerView = findViewById(R.id.recyclerView);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));

        pokeApiService = RetrofitClient.getClient("https://pokeapi.co/api/v2/").create(PokeApiService.class);
        fetchPokemonList();
    }

    private void fetchPokemonList() {
        Call<PokemonResponse> call = pokeApiService.getPokemonList();
        call.enqueue(new Callback<PokemonResponse>() {
            @Override
            public void onResponse(Call<PokemonResponse> call, Response<PokemonResponse> response) {
                if (response.isSuccessful()) {
                    PokemonResponse pokemonResponse = response.body();
                    List<Pokemon> pokemonList = pokemonResponse.getPokemonList();
                    adapter = new PokemonListAdapter(MainActivity.this, pokemonList);
                    recyclerView.setAdapter(adapter);
                } else {
                    // Handle API error
                }
            }

            @Override
            public void onFailure(Call<PokemonResponse> call, Throwable t) {
                // Handle network error
            }
        });
    }
}

File: app/src/main/java/com/nickand/pokemondex/MainActivity.kt
package com.nickand.pokemondex

import androidx.appcompat.app.AppCompatActivity
import com.nickand.pokemondex.PokeApiService
import androidx.recyclerview.widget.RecyclerView
import com.nickand.pokemondex.PokemonListAdapter
import android.os.Bundle
import com.nickand.pokemondex.R
import androidx.recyclerview.widget.LinearLayoutManager
import com.nickand.pokemondex.RetrofitClient
import com.nickand.pokemondex.PokemonResponse
import com.nickand.pokemondex.Pokemon
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class MainActivity : AppCompatActivity() {
    private lateinit var pokeApiService: PokeApiService
    private var recyclerView: RecyclerView? = null
    private var adapter: PokemonListAdapter? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        recyclerView = findViewById(R.id.recyclerView)
        recyclerView?.let {
            it.layoutManager = LinearLayoutManager(this)
        }
        pokeApiService = RetrofitClient.getClient("https://pokeapi.co/api/v2/")?.create(
            PokeApiService::class.java
        )!!
        fetchPokemonList()
    }

    private fun fetchPokemonList() {
        val call = pokeApiService.pokemonList
        call?.enqueue(object : Callback<PokemonResponse?> {
            override fun onResponse(
                call: Call<PokemonResponse?>,
                response: Response<PokemonResponse?>
            ) {
                if (response.isSuccessful) {
                    val pokemonResponse = response.body()
                    val pokemonList = pokemonResponse?.pokemonList
                    adapter = PokemonListAdapter(this@MainActivity, pokemonList ?: emptyList())
                    recyclerView?.adapter = adapter
                } else {
                    // Handle API error
                }
            }

            override fun onFailure(call: Call<PokemonResponse?>, t: Throwable) {
                // Handle network error
            }
        })
    }
}